### PR TITLE
ref(react): change mixins -> hocs [wip]

### PR DESCRIPTION
**Pull request by billyvg**
example if we wanted to try to get rid of mixins usage

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/static/sentry/app/providers/withOrganization.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+
+import SentryTypes from '../proptypes';
+
+export default function withOrganization(WrappedComponent) {
```
Comment (benvinegar) [inline src/sentry/static/sentry/app/providers/withOrganization.jsx]: I personally think we should drop these "context" mixins/HOCs. I don't like that they introduce magic prop functions to access properties like `access` and `features`. 

I think it's easier and more clear for child components to just specify `contextTypes` and opt-in where appropriate. If we want utility methods like `getAccess`, let's declare them on the `organization` object itself when it's initialized (e.g. wrap the object in a utility class/object).

Comment (benvinegar) [reply src/sentry/static/sentry/app/providers/withOrganization.jsx]: I guess one benefit is that we could move these values to stores, and not go through `contextTypes` at all ... which could avoid some of the janky context propagation problems that can occur.

**Comment by castlec** [comment]
@benvinegar Have you formed a more solid opinion on this PR? Should it be updated to fix conflicts or rejected?

**Comment by billyvg** [comment]
Going to close this in favor of simpler HoCs (e.g. just providing `organization` from a source (currently from `context`)).

**Pull request closed without merging**